rmgpy/rmgobject.pyx

```diff
@@ -133,3 +134,52 @@ cpdef expand_to_dictionaries(obj):
             return expand_to_dictionaries(obj.as_dict())
         else:
             return obj
+
+cpdef recursive_make_object(obj, class_dictionary, make_final_object=True):
+    if isinstance(obj, dict):
+        if 'class' in obj:  # This is a dictionary of an object to be created
+            try:
+                class_to_make = class_dictionary[obj['class']]
+            except KeyError:
+                raise KeyError('Class {0} must be provided in the class_dictionary: {1}'.format(obj['class'],
+                                                                                                class_dictionary))
+
+            for key, val in obj.iteritems():
+                if key == 'class':
```

Comment:
maybe change `==` to `!=` and skip the `continue`?

Reply:
Thanks for the catch!